bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java

```diff
@@ -226,13 +238,14 @@ public BigtableBufferedMutator(
       ExceptionListener exceptionListener,
       String host,
       int maxInflightRpcs,
-      TableName tableName) {
+      TableName tableName,
+      ExecutorService heapSizeExecutor) {
     this.batchExecutor = batchExecutor;
     this.configuration = null;
     this.exceptionListener = exceptionListener;
     this.host = host;
     this.tableName = tableName;
-    this.sizeManager = new HeapSizeManager(maxHeapSize, maxInflightRpcs);
+    this.sizeManager = new HeapSizeManager(maxHeapSize, maxInflightRpcs, heapSizeExecutor);
```

Comment:
Can one constructor be reused, so that we don't have to repeat the code like instantiating HeapSizeManager?